chain/chain/src/flat_storage_resharder.rs

```diff
@@ -1221,701 +890,240 @@ fn is_flat_head_on_par_with_chain(
-            StateChangeCause,
-        },
-    };
-    use near_store::{
-        flat::{BlockInfo, FlatStorageManager, FlatStorageReadyStatus},
-        genesis::initialize_genesis_state,
-        test_utils::create_test_store,
+    use near_primitives::hash::CryptoHash;
+    use near_primitives::shard_layout::ShardLayout;
+    use near_primitives::state::FlatStateValue;
+    use near_primitives::trie_key::TrieKey;
+    use near_primitives::types::{AccountId, ShardId};
+    use near_store::ShardUId;
+    use near_store::adapter::StoreAdapter;
+    use near_store::adapter::flat_store::FlatStoreAdapter;
+    use near_store::flat::{
+        BlockInfo, FlatStorageReadyStatus, FlatStorageReshardingStatus, FlatStorageStatus,
+        ParentSplitParameters,
     };
+    use near_store::genesis::initialize_genesis_state;
+    use near_store::test_utils::create_test_store;
 
-    use crate::{
-        Chain, ChainGenesis, DoomslugThresholdMode, rayon_spawner::RayonAsyncComputationSpawner,
-        runtime::NightshadeRuntime, types::ChainConfig,
-    };
+    use crate::flat_storage_resharder::FlatStorageReshardingTaskResult;
+    use crate::rayon_spawner::RayonAsyncComputationSpawner;
+    use crate::runtime::NightshadeRuntime;
+    use crate::types::{ChainConfig, RuntimeAdapter};
+    use crate::{Chain, ChainGenesis, DoomslugThresholdMode};
 
-    use super::*;
-    use assert_matches::assert_matches;
-    use more_asserts::assert_gt;
-    use near_async::messaging::{CanSend, IntoMultiSender};
-    use near_crypto::{KeyType, PublicKey};
-
-    /// Shorthand to create account ID.
-    macro_rules! account {
-        ($str:expr) => {
-            $str.parse::<AccountId>().unwrap()
-        };
-    }
+    use super::FlatStorageResharder;
 
-    trait TestSender:
-        CanSend<FlatStorageSplitShardRequest>
-        + CanSend<FlatStorageShardCatchupRequest>
-        + CanSend<MemtrieReloadRequest>
-    {
-        fn new(chain_store: ChainStore) -> Self;
+    /// Simple shard layout with two shards.
+    fn simple_shard_layout() -> ShardLayout {
+        let s0 = ShardId::new(0);
+        let s1 = ShardId::new(1);
+        let shards_split_map = BTreeMap::from([(s0, vec![s0]), (s1, vec![s1])]);
+        ShardLayout::v2(vec!["ff".parse().unwrap()], vec![s0, s1], Some(shards_split_map))
     }
 
-    /// Simple sender to execute tasks immediately on the same thread where they are issued.
-    struct SimpleSender {
-        chain_store: Arc<Mutex<ChainStore>>,
+    /// Derived from [simple_shard_layout] by splitting the second shard.
+    fn shard_layout_after_split() -> ShardLayout {
+        ShardLayout::derive_shard_layout(&simple_shard_layout(), "pp".parse().unwrap())
     }
 
-    impl TestSender for SimpleSender {
-        fn new(chain_store: ChainStore) -> Self {
-            Self { chain_store: Arc::new(Mutex::new(chain_store)) }
+    fn setup_test() -> (FlatStorageResharder, ParentSplitParameters) {
```

Comment:
This function is actually almost the same as `create_chain_resharder_sender`. Not sure why didn't the diff group these together :/

Reply:
It's much better if you look here https://github.com/near/nearcore/pull/13525/commits/d9dc17c5ae3da771bb9834c1ab29cd18144570ca